metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg_profiler.py

```diff
@@ -109,92 +113,101 @@ def profile_table(
+                    "profiling",
+                    f"Error while profiling dataset {dataset_name}: {e}",
+                )
+            if row_count:
+                # Iterating through fieldPaths introduces unwanted stats for list element fields...
+                for field_path, field_id in table.schema()._name_to_id.items():
+                    field = table.schema().find_field(field_id)
+                    column_profile = DatasetFieldProfileClass(fieldPath=field_path)
                     if self.config.include_field_null_count:
-                        null_counts = self._aggregate_counts(
-                            null_counts, data_file.null_value_counts
+                        column_profile.nullCount = cast(
+                            int, null_counts.get(field_id, 0)
+                        )
+                        column_profile.nullProportion = float(
+                            column_profile.nullCount / row_count
                         )
+
                     if self.config.include_field_min_value:
-                        self._aggregate_bounds(
-                            table.schema(),
-                            min,
-                            min_bounds,
-                            data_file.lower_bounds,
+                        column_profile.min = (
+                            self._render_value(
+                                dataset_name, field.field_type, min_bounds.get(field_id)
+                            )
+                            if field_id in min_bounds
+                            else None
                         )
                     if self.config.include_field_max_value:
-                        self._aggregate_bounds(
-                            table.schema(),
-                            max,
-                            max_bounds,
-                            data_file.upper_bounds,
+                        column_profile.max = (
+                            self._render_value(
+                                dataset_name, field.field_type, max_bounds.get(field_id)
+                            )
+                            if field_id in max_bounds
+                            else None
                         )
-                    total_count += data_file.record_count
-        except Exception as e:
-            # Catch any errors that arise from attempting to read the Iceberg table's manifests
-            # This will prevent stateful ingestion from being blocked by an error (profiling is not critical)
-            self.report.report_warning(
-                "profiling",
-                f"Error while profiling dataset {dataset_name}: {e}",
+                    dataset_profile.fieldProfiles.append(column_profile)
+            time_taken = timer.elapsed_seconds()
+            self.report.report_table_profiling_time(time_taken)
+            LOGGER.debug(
+                "Finished profiling of dataset: %s in %s", dataset_name, time_taken
```

Comment:
please us fstring instead of string formatting

Reply:
Done